modules/space/rendering/renderablehabitablezone.cpp

```diff
@@ -0,0 +1,278 @@
+
+    glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() * modelTransform;
+
+    _shader->setUniform(
+        _uniformCache.modelViewProjection,
+        data.camera.projectionMatrix() * glm::mat4(modelViewTransform)
+    );
+    _shader->setUniform(_uniformCache.width, _width);
+    _shader->setUniform(_uniformCache.opacity, _opacity);
+    _shader->setUniform(_uniformCache.conservativeBounds, _conservativeBounds);
+    _shader->setUniform(_uniformCache.showOptimistic, _showOptimistic);
+
+    ghoul::opengl::TextureUnit unit;
+    unit.activate();
+    _texture->bind();
+    _shader->setUniform(_uniformCache.texture, unit);
+
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glDepthMask(false);
+    glDisable(GL_CULL_FACE);
+
+    _plane->render();
+
+    _shader->deactivate();
+
+    // Restores GL State
+    global::renderEngine->openglStateCache().resetBlendState();
+    global::renderEngine->openglStateCache().resetDepthState();
+    global::renderEngine->openglStateCache().resetPolygonAndClippingState();
+}
+
+void RenderableHabitableZone::initializeShader() {
+    _shader = global::renderEngine->buildRenderProgram(
+        "HabitableZoneProgram",
+        absPath("${MODULE_SPACE}/shaders/habitablezone_vs.glsl"),
+        absPath("${MODULE_SPACE}/shaders/habitablezone_fs.glsl")
+    );
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+}
+
+void RenderableHabitableZone::updateUniformLocations() {
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+}
+
+void RenderableHabitableZone::computeZone() {
+    glm::vec4 distancesInAu = computeKopparapuZoneBoundaries(_teff, _luminosity);
+    constexpr float AU = static_cast<float>(distanceconstants::AstronomicalUnit);
+    const float inner = distancesInAu[0] * AU;
+    const float innerConservative = distancesInAu[1] * AU;
+    const float outerConservative = distancesInAu[2] * AU;
+    const float outer = distancesInAu[3] * AU;
+
+    float discWidth = 0.f;
+    if (outer > 0.f) {
+        discWidth = (outer - inner) / outer;
+    }
+
+    _size.set(outer);
+    _width.set(discWidth);
+
+    // Compute the coservative bounds normalized by the size of the disc, i.e. in [0, 1]
+    _conservativeBounds = glm::vec2(innerConservative, outerConservative);
+    _conservativeBounds /= _size;
+}
+
+glm::vec4 RenderableHabitableZone::computeKopparapuZoneBoundaries(float teff,
+                                                                  float luminosity)
+{
+    // Kopparapu's formula only considers stars with teff in range [2600, 7200] K.
+    // However, we want to use the formula for more stars, so add some flexibility to
+    // the teff boundaries.
+    // OBS! This also prevents problems with too large values in the distance computation
+    if (teff > 8000.f || teff < 2000.f) {
+        // For the other stars, use a method by Tom E. Morris:
```

Comment:
Is this something we can communicate to the user through the interface? I.e a property that is changed by  the renderable so they know? Is this not worth it b/c it will wind up looking the same? Thinking if some will look different then others, it might be nice to explain why.

Reply:
That's a good point. A drawback with this method is the we don't get the optimistic boundaries, which will affect the appearance of the disc depending on the transfer function/texture. I think a property would make sense, good suggestion!